chocolatey/choco
# (#2597) Add Pester tests to test the package built

## Description Of Changes

This PR brings out tests and packages from Test Kitchen into the GitHub repository. This will allow a developer to build the software locally.

## Motivation and Context

Allowing developers to test their changes don't break other established scenarios.

## Testing

At this point all testing has been done on a fairly fresh Windows 10 VM.

1. Install prerequisite PowerShell module Pester: `choco install pester --version 5.3.1`
2. Build the Chocolatey source code using the `./build.bat` script.
3. In an elevated PowerShell prompt, run `./Invoke-Tests.ps1`
4. Ensure that all tests are passing. - There is a known issue (that will get an issue opened for it) Where the test `choco install.Force Installing a Package that is already installed (with a read/delete locked file).Should not have been able to delete the rollback` fails under Windows 10/11, but not under Server 2016 or 2019. All other tests are passing, and this appears to be something with the way the client OS handles things different from a server OS.

Another way this can be tested is with the Vagrantfile put in the `tests` directory. From the `tests` directory, if you run `vagrant up` it will stand up a VM, install the necessary software, then build and test Chocolatey (it will also test it as though it is on Team City which means all of the signing tests will fail)

## Change Types Made

* [x] PowerShell code changes. (Pester tests were added).

## Related Issue

Fixes #2597 

## Change Checklist

* [ ] ~~PowerShell v2 compatibility checked.~~ PowerShell v2 compatibility has not been checked as these tests require Pester 5.3.1 which requires a minimum of PowerShell v3. All tests have been tested in a Windows 10 environment with PowerShell 5.1
